server/migrate/proto/migrate.proto

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright (C) 2020 Grakn Labs
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+// A migration service protocol to activate and monitor migrations remotely.
+// Backwards compatibility is currently not enforced on this protocol
+
+syntax = "proto3";
+
+option java_package = "grakn.core.server.migrate.proto";
+option java_outer_classname = "MigrateProto";
+
+import "server/migrate/proto/data.proto";
+
+package migrate;
+
+service MigrateService {
+    rpc exportFile (ExportFile.Req) returns (stream Job.Res);
+    rpc importFile (ImportFile.Req) returns (stream Job.Res);
+    rpc exportStream (ExportStream.Req) returns (stream data.Item);
+    rpc exportSchema (ExportSchema.Req) returns (ExportSchema.Res);
+}
+
+message ExportFile {
+    message Req {
+        string name = 1;
+        string path = 2;
+    }
+}
+
+message Job {
+    message Res {
+        oneof res {
+            Progress progress = 1;
+            Completion completion = 2;
+        }
+    }
+
+    message Progress {
+        uint64 total_count = 1;
+        uint64 current_progress = 2;
+    }
+
+    message Completion {
+        uint64 total_count = 1;
+    }
+}
+
+message ImportFile {
+    message Req {
+        string name = 1;
+        string path = 2;
+        map<string, string> remap_labels = 3;
+    }
+}
+
+message ExportStream {
+    message Req {
+        string name = 1;
+    }
+}
+
+message ExportSchema {
+    message Req {
+        string name = 1;
+    }
+
+    message Res {
+        string schema = 1;
```

Comment:
Might be worth noting that there is a string limit size of i think 4mb messages? I think this might be about 4 million characters uncompressed, hopefully we don't actually hit this limit at any point!

Reply:
I highly doubt it, this should easily support schemas well into the 10's of thousands of schema concepts.